test/metabase/upload_test.clj

```diff
@@ -258,16 +268,26 @@
                                "Rey Skywalker, 170, 15"
                                "Darth Vader, 202.0, 41.9BBY"])))))
     (testing "Boolean coalescing"
-      (is (=? (with-ai-id {:name          vchar-type
-                           :is_jedi_      bool-type
-                           :is_jedi__int_ int-type
-                           :is_jedi__vc_  vchar-type})
+      (is (=? (with-ai-id {:name                    vchar-type
+                           :is_jedi_                bool-type
+                           :is_jedi__int_and_bools_ vchar-type
+                           :is_jedi__vc_            vchar-type})
               (@#'upload/detect-schema
-               (csv-file-with ["Name, Is Jedi?, Is Jedi (int), Is Jedi (VC)"
-                               "Rey Skywalker, yes, true, t"
-                               "Darth Vader, YES, TRUE, Y"
-                               "Grogu, 1, 9001, probably?"
-                               "Han Solo, no, FaLsE, 0"])))))
+               (csv-file-with ["         Name, Is Jedi?, Is Jedi (int and bools), Is Jedi (VC)"
+                               "Rey Skywalker,      yes,                    true,            t"
+                               "  Darth Vader,      YES,                    TRUE,            Y"
+                               "        Grogu,        1,                    9001,    probably?"
+                               "     Han Solo,       no,                   FaLsE,            0"])))))
+    (testing "Boolean and integers together"
+      (is (=? (with-ai-id {:vchar       vchar-type
+                           :bool        bool-type
+                           :bool_or_int bool-type})
+              (@#'upload/detect-schema
+               (csv-file-with ["vchar,bool,bool-or-int"
+                               " true,true,          1"
+                               "    1,   1,          0"
+                               "    2,   0,          0"
+                               "   no,  no,          1"])))))
```

Comment:
Would be nice to see another column like 1/0/1/2 to check it correctly converges to int.

Reply:
Good idea, added this to both the new tests ba2caccb11807b0d531a8e32977135df934d0814